### DISCUSSION TITLE
Add ability create initial XTCE documents

### DISCUSSION THREAD

**Discussion by greglucas** [lasp/space_packet_parser#74] [Ideas]
It would be useful to provide utilities to create XTCE documents from other formats. For example, from various csv / spreadsheet definitions. This would enable XTCE to be "the" format we accept, and provide the ability to move between the various formats outside of the parsing logic which can always be XTCE-based.

On IMAP, we have an Excel -> XTCE CLI tool that works for that mission's packet definitions.
https://github.com/IMAP-Science-Operations-Center/imap_processing/blob/dev/imap_processing/ccsds/excel_to_xtce.py

We could bring this functionality over to `space_packet_parser` for others to use, and potentially add more supported formats like our cubesat missions as well.

Questions:
* Is this even a good idea, or should we try to load the definitions directly from what the users provide (this would add (potentially optional) dependencies on excel readers etc...)
* Do we want a CLI tool for this? (Excel -> XTCE, csv -> XTCE, ...)

**Comment by medley56** [comment]
I think we want the ability to convert from a CSV definition to XTCE so we can use the rest of the codebase as-is. This converter should have both a CLI interface (minimal) and a programmatic API that converts to an in-memory XTCE document and feeds it directly into an XtcePacketDefinition object, effectively allowing us to create an XtcePacketDefinition from a CSV. Basically an adapter. 

In terms of the CSV format, I need input from @mchambli and @greglucas on exactly what format(s) we should support.

Any dependencies required for this CSV --> XTCE converter should be declared as an extra in our dependency spec so that people can install with (e.g.) `pip install space-packet-parser[csv]`
